lib/msf/core/exploit/remote/http/http_cookie_jar.rb

```diff
@@ -59,6 +59,12 @@ def parse_and_merge(set_cookie_header, origin_url, options = nil)
             parsed_cookies.each { |c| add(Msf::Exploit::Remote::HTTP::HttpCookie.new(c)) }
             parsed_cookies
           end
+
+          # Modules are replicated before running. This method ensures that the cookie jar from
+          # one run, will not impact subsequent runs.
+          def initialize_copy(other)
+            @cookie_jar = other.instance_variable_get(:@cookie_jar).clone
+          end
```

Comment:
I'm not very familiar with `initialize_copy` mechanism, but is it okay to implement it without calling `super` to make sure any possible other implementation will be called?

Reply:
initialize_copy is defined natively in C:
https://github.com/ruby/ruby/blob/b655a3fa5b8d1a30565e19425c28a1cfd8631165/object.c#L4590

Implementation:
https://github.com/ruby/ruby/blob/b655a3fa5b8d1a30565e19425c28a1cfd8631165/object.c#L611-L628

Looks like this behavior of `super` isn't specifically needed, but I'll add it in as it shouldn't break anything - but either way there's unit tests to catch any future Ruby 4.x changes in behavior🤞

Reply:
Awesome! Thanks for the explanation!